spec-dtslint/observables/fromEvent-spec.ts

```diff
@@ -0,0 +1,20 @@
+import { fromEvent } from 'rxjs';
+import { NodeStyleEventEmitter, HasEventTargetAddRemove } from '../../src/internal/observable/fromEvent';
+
+const hasEventTargetAddRemove: HasEventTargetAddRemove<string> = {
+  addEventListener: (type, listener, options?) => { },
+  removeEventListener: (type, listener, options?) => { }
+};
+
+const nodeStyleEventEmitter: NodeStyleEventEmitter = {
+  addListener: (eventName, handler) => nodeStyleEventEmitter ,
+  removeListener: (eventName, handler) => nodeStyleEventEmitter
+};
+
+it('should infer correctly with HasEventTargetAddRemove event type', () => { 
+  const a = fromEvent(hasEventTargetAddRemove, 'foo'); // $ExpectType Observable<string>
+});
+
+it('should infer correctly with NodeStyleEventEmitter event type', () => {
```

Comment:
Hi @cartant,

Am I already in the right direction for this test?

Reply:
Sorry to ping you again @cartant. 
I was just wondering if this is a need-to-fix typing or I am doing something wrong here to test the inference?'
or maybe @benlesh could help explaning?

Reply:
I think this looks okay... However, we probably want to test it against "real world" things... Try creating a DOMElement type like:   `const div: HTMLDivElement = !null;`  and seeing what you get back from that. You could do the same for node things like `process`, for example. Remember, it doesn't have to work at run time, it just needs to infer the correct type.